pkg/apiserver/resthandler.go

```diff
@@ -488,8 +488,7 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		gv := scope.Kind.GroupVersion()
 		codec := runtime.NewCodec(
 			scope.Serializer.EncoderForVersion(s, gv),
-			scope.Serializer.DecoderToVersion(s, unversioned.GroupVersion{Group: gv.Group, Version: runtime.APIVersionInternal}),
-		)
+			scope.Serializer.DecoderToVersion(s, gv))
```

Comment:
No, I checked the client code and it doesn't use internal objects at all. What uses them & causes the behavior observed in #25106 (internal field names slipping in) is that the e2e test mentioned in #25106 uses direct unmarshalling of v1 json into internal objects [here](https://github.com/kubernetes/kubernetes/blob/700fbd0543e38c610c124ae55e49039b2b34cca9/test/e2e/kubectl.go#L1308)